pkg/operator/staticpod/certsyncpod/certsync_cmd.go

```diff
@@ -50,36 +55,65 @@ func NewCertSyncControllerCommand(configmaps, secrets []revision.RevisionResourc
 	return cmd
 }
 
-func (o *CertSyncControllerOptions) Complete() (*CertSyncController, error) {
-	kubeConfig, err := client.GetKubeConfigOrInClusterConfig(o.KubeConfigFile, nil)
+func (o *CertSyncControllerOptions) Run() error {
+	// When the kubeconfig content change, commit suicide to reload its content.
+	observer, err := fileobserver.NewObserver(500 * time.Second)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	protoKubeConfig := rest.CopyConfig(kubeConfig)
-	protoKubeConfig.AcceptContentTypes = "application/vnd.kubernetes.protobuf,application/json"
-	protoKubeConfig.ContentType = "application/vnd.kubernetes.protobuf"
 
-	// This kube client use protobuf, do not use it for CR
-	kubeClient, err := kubernetes.NewForConfig(protoKubeConfig)
-	if err != nil {
-		return nil, err
-	}
-	kubeInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, 10*time.Minute, informers.WithNamespace(o.Namespace))
+	initialContent, _ := ioutil.ReadFile(o.KubeConfigFile)
+	observer.AddReactor(fileobserver.ExitOnChangeReactor, map[string][]byte{o.KubeConfigFile: initialContent}, o.KubeConfigFile)
+
+	stopCh := make(chan struct{})
+	go observer.Run(stopCh)
 
-	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events(o.Namespace), "cert-syncer",
+	kubeInformers := informers.NewSharedInformerFactoryWithOptions(o.kubeClient, 10*time.Minute, informers.WithNamespace(o.Namespace))
+	go kubeInformers.Start(stopCh)
+
+	eventRecorder := events.NewKubeRecorder(o.kubeClient.CoreV1().Events(o.Namespace), "cert-syncer",
 		&corev1.ObjectReference{
 			APIVersion: "v1",
 			Kind:       "Pod",
 			Namespace:  os.Getenv("POD_NAMESPACE"),
 			Name:       os.Getenv("POD_NAME"),
 		})
 
-	return NewCertSyncController(
+	controller, err := NewCertSyncController(
 		o.DestinationDir,
 		o.Namespace,
 		o.configMaps,
 		o.secrets,
 		kubeInformers,
 		eventRecorder,
 	)
+	if err != nil {
+		return err
+	}
+	go controller.Run(1, stopCh)
```

Comment:
without go routine?

Reply:
i'm waiting for the stopCh to close, I think that is equal to what you suggests? (+I can print "this is shutting down)

Reply:
yes, it's the same. Just one line less :)